src/ModelObjects/representations.rs

```diff
@@ -150,15 +150,18 @@ impl<'a> SystemRepresentation {
                     || right_side.collect_open_edges(states, index, action, open_edges, sync_type)
             }
             SystemRepresentation::Conjunction(left_side, right_side) => {
-                let open_edges_len = open_edges.len();
-                if left_side.collect_open_edges(states, index, action, open_edges, sync_type) {
-                    let left_found_transitions = open_edges_len != open_edges.len();
-                    if right_side.collect_open_edges(states, index, action, open_edges, sync_type) {
-                        let right_found_transitions = open_edges_len != open_edges.len();
-                        return left_found_transitions == right_found_transitions;
-                    }
+                let mut left = vec![];
+                let mut right = vec![];
+
+                left_side.collect_open_edges(states, index, action, &mut left, sync_type);
+                right_side.collect_open_edges(states, index, action, &mut right, sync_type);
+
+                if !(left.is_empty() || right.is_empty()) {
+                    open_edges.append(&mut left);
+                    open_edges.append(&mut right);
                 }
-                false
+
+                true
```

Comment:
This was the change that solved the tests.

This basically just removes the crash that would occur when returning false. Instead we only add the conjunction transition if both sub components have edges.

The error shouldn't happened as S refines T would crash when S had an output which only one of 2 subcomponents have matching edges for. 

Note that these edges do not have to be open to be matched, as that check is handled later.